### PR TITLE
Metadata for sharing lobby

### PR DESCRIPTION
**Pull request by skick1234**
We can add more properties such as language to `Lobby` struct.

### CODE REVIEW

**Comment by Bios-Marcel** [comment]
Instead of displaying the static properties of the lobby, wouldn't it be more interesting to report the current progress?

E.g. "5/12 players and round 1 of 5" or something like that?

File: communication/templates/robot.html [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    <meta property="og:title" content="Scribble.rs - Join Lobby">
+    <meta property="og:description" content="{{.MaxPlayers}} Players - {{.MaxRounds}} Rounds">
+    <meta property="og:image" content="/resources/favicon.png">
+
+    <!-- Twitter -->
+    <meta property="twitter:card" content="summary_large_image">
+    <meta property="twitter:title" content="Scribble.rs - Join Lobby">
+    <meta property="twitter:description" content="{{.MaxPlayers}} Players - {{.MaxRounds}} Rounds">
+    <meta property="twitter:image" content="/resources/favicon.png">
+    <link rel="stylesheet" type="text/css" href="/resources/base.css" />
```
Comment (Bios-Marcel) [inline communication/templates/robot.html:23]: `@font.face` declaration is missing. Has this been left out intentionally?
If so, we might want to add a comment as to why these aren't here, but for example on the error page.

**Comment by skick1234** [comment]
> Instead of displaying the static properties of the lobby, wouldn't it be more interesting to report the current progress?
> 
> E.g. "5/12 players and round 1 of 5" or something like that?

Caching problem: Robots don't fetch the same url many times.

File: communication/templates/robot.html (hunk 1 above)
Comment (skick1234) [reply communication/templates/robot.html:23]: I copied from the old `error.html`. It's missing `{{.RootPath}}` I think. I'll check it.

**Comment by Bios-Marcel** [comment]
> Caching problem: Robots don't fetch the same url many times.

I mean, people mostly share this live, so I assume it'd only be interesting for the moment anyways.
It's not like the google webscraper will scrape this, as it's not even indexed.
A usecase I think we could test it discord for example and see how frequently, if not always, it updates.

File: communication/templates/robot.html [hunk 2]
```diff
@@ -6,19 +6,19 @@
...
+    <meta name="description" content="{{.EditableLobbySettings.MaxPlayers}} Players - {{.MaxRounds}} Rounds">
     <meta property="og:type" content="website">
     <meta property="og:title" content="Scribble.rs - Join Lobby">
-    <meta property="og:description" content="{{.MaxPlayers}} Players - {{.MaxRounds}} Rounds">
+    <meta property="og:description" content="{{.EditableLobbySettings.MaxPlayers}} Players - {{.MaxRounds}} Rounds">
     <meta property="og:image" content="{{.RootPath}}/resources/favicon.png">
     <meta property="twitter:card" content="summary_large_image">
     <meta property="twitter:title" content="Scribble.rs - Join Lobby">
-    <meta property="twitter:description" content="{{.MaxPlayers}} Players - {{.MaxRounds}} Rounds">
+    <meta property="twitter:description" content="{{.EditableLobbySettings.MaxPlayers}} Players - {{.MaxRounds}} Rounds">
```
Comment (Bios-Marcel) [inline communication/templates/robot.html:16]: This should not be necessary, as it's an anonymous struct, e.g. the fields are treated as if they are in the struct that `EditableLobbySettings` has been embedded in. Did you get an error here?

Comment (skick1234) [reply communication/templates/robot.html]: > This should not be necessary, as it's an anonymous struct, e.g. the fields are treated as if they are in the struct that `EditableLobbySettings` has been embedded in. Did you get an error here?

I get an error on the first try and I realize that it's come from your refactoring struct. :v

File: communication/templates/lobby.html [hunk 3]
```diff
@@ -728,7 +728,7 @@
             const hsp = Math.sqrt(
                 0.299 * (innerColor[0] * innerColor[0]) +
                 0.587 * (innerColor[1] * innerColor[1]) +
-                0.114 * (innerColor[2] * innerColor[2]);
+                0.114 * (innerColor[2] * innerColor[2])
```
Comment (skick1234) [inline communication/templates/lobby.html:731]: I forgot to comment that but it's a bug :v

File: communication/templates/robot.html (hunk 2 above)
Comment (skick1234) [reply communication/templates/robot.html]: Removed!

File: communication/templates/lobby.html [hunk 4]
```diff
@@ -728,7 +728,7 @@
             const hsp = Math.sqrt(
                 0.299 * (innerColor[0] * innerColor[0]) +
                 0.587 * (innerColor[1] * innerColor[1]) +
-                0.114 * (innerColor[2] * innerColor[2]);
```
Comment (Bios-Marcel) [inline communication/templates/lobby.html:731]: good catch, I've added that by accident ^^

**Comment by Bios-Marcel** [comment]
Ready to merge, right?

**Comment by skick1234** [comment]
> > Caching problem: Robots don't fetch the same url many times.
> 
> I mean, people mostly share this live, so I assume it'd only be interesting for the moment anyways.
> It's not like the google webscraper will scrape this, as it's not even indexed.
> A usecase I think we could test it discord for example and see how frequently, if not always, it updates.

I know, but it's not really a `live info`. And I just add this because I don't want to see `Scribble.rs - Error` title :>

**Comment by skick1234** [comment]
> Ready to merge, right?

Ready :>

**Comment by Bios-Marcel** [comment]
It's really misleading imo. Now it looks like there's 12 players in a lobby that's empty :thinking: 
Guess I'll change it a bit.

**Comment by skick1234** [comment]
You mean it should be `max players` and `max rounds`. But I don't like it at all :v it's just my preference

**Pull request merged**